lib/chef/provider/service/macosx.rb

```diff
@@ -77,6 +95,11 @@ def define_resource_requirements
               @current_resource.running(false)
             end
           end
+
+          requirements.assert(:all_actions) do |a|
+            a.assertion { @plist.nil? ? true : ::File.exists?(@plist) }
+            a.failure_message Chef::Exceptions::Service, "#{@plist} does not exist"
+          end
```

Comment:
Meh. Unfortunately we try to load the plist in #find_service_label via #load_current_resource, before #define_resource_requirements. This seems like a super common/sad gotcha. @chef/client-core?

```
Generated at 2015-03-26 20:24:28 -0400
Mixlib::ShellOut::ShellCommandFailed: macosx_service[com.apple.ZoomWindow] ((chef-apply cookbook)::(chef-apply recipe) line 1) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of plutil -convert xml1 -o - does_not_exist ----
STDOUT: does_not_exist: file does not exist or is not readable or is not a regular file
STDERR: 
---- End output of plutil -convert xml1 -o - does_not_exist ----
Ran plutil -convert xml1 -o - does_not_exist returned 1
/Users/btm/.rvm/gems/ruby-2.1.4/gems/mixlib-shellout-2.0.1/lib/mixlib/shellout.rb:278:in `invalid!'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/mixlib-shellout-2.0.1/lib/mixlib/shellout.rb:265:in `error!'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/mixin/shell_out.rb:66:in `shell_out_with_systems_locale!'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/provider/service/macosx.rb:220:in `find_service_label'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/provider/service/macosx.rb:51:in `load_current_resource'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/provider.rb:128:in `run_action'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/resource.rb:561:in `run_action'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/runner.rb:49:in `run_action'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/runner.rb:81:in `block (2 levels) in converge'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/runner.rb:81:in `each'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/runner.rb:81:in `block in converge'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/runner.rb:80:in `converge'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/application/apply.rb:166:in `run_chef_recipe'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/application/apply.rb:175:in `run_application'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/lib/chef/application/apply.rb:188:in `run'
/Users/btm/.rvm/gems/ruby-2.1.4/gems/chef-12.2.0.dev.0/bin/chef-apply:25:in `<top (required)>'
/Users/btm/.rvm/gems/ruby-2.1.4/bin/chef-apply:23:in `load'
/Users/btm/.rvm/gems/ruby-2.1.4/bin/chef-apply:23:in `<main>'
/Users/btm/.rvm/gems/ruby-2.1.4/bin/ruby_executable_hooks:15:in `eval'
/Users/btm/.rvm/gems/ruby-2.1.4/bin/ruby_executable_hooks:15:in `<main>'
```